File: nodes/array-index-selector/array-index-selector.process.py
async def process(inputs, settings, config, nodeConfig):
    """
    Process function for the Array Index Selector node.
    """
    array = inputs.get("array", [])
    index = inputs.get("index", 0)

    index_to_use = index

    # Check if index is a number and round it
    if isinstance(index_to_use, (int, float)):
        index_to_use = round(index_to_use)
    
    # Safely handle array index access
    element = None
    if 0 <= index_to_use < len(array):
        element = array[index_to_use]

    # Raise an error if the element is undefined (out of bounds)
    if element is None:
        raise IndexError(f"Index {index_to_use} is out of bounds for array of length {len(array)}")

    return {"element": element}

File: nodes/array-index-selector/array-index-selector.process.js
export default async ({inputs, settings, config}) => {

  const { array, index } = inputs;

  let indexToUse = index;

  if(typeof indexToUse === "number") {
    // Round index to nearest integer
    indexToUse = Math.round(index);
  }

  // Safely handle array index access
  const element = array[indexToUse] === undefined ? undefined : array[indexToUse];

  if(element === undefined) {
    throw new Error(`Index ${indexToUse} is out of bounds for array of length ${array.length}`);
  }

  return { element };
};
